Reference/Events/Application-Startup.md

```diff
@@ -88,51 +88,18 @@ If you want more control over execution you can override these properties:
-* `Complete` --> `ApplicationStarted`
+A custom weight can be assigned to a handler by marking the class with the [WeightAttribute](../../apidocs/csharp/api/Umbraco.Core.ObjectResolution.WeightAttribute.html). Positive weights are considered "user-space" while negative weights are "core". 
 
-## Binding to HttpApplication events
+For example, the [CacheRefresherEventHandler](../../apidocs/csharp/api/Umbraco.Web.Cache.CacheRefresherEventHandler.html) is marked with `Weight(int.MinValue)` because its events need to run before anything else. 
 
-Umbraco allows you to bind directly to HttpApplication events which is very handy since normally you would require an HttpModule to bind to these types of events. The HttpApplication events are listed here: [http://msdn.microsoft.com/en-us/library/system.web.httpapplication_events(v=vs.110).aspx](http://msdn.microsoft.com/en-us/library/system.web.httpapplication_events(v=vs.110).aspx)
+Finally (very EXPERT), users can [register a filter](../../apidocs/csharp/api/Umbraco.Core.ObjectResolution.ApplicationEventsResolver.html#Umbraco_Core_ObjectResolution_ApplicationEventsResolver_FilterCollection) to process the list (after it has been ordered) and re-order it, or remove handlers.
 
-In order to bind to these events you need to first listen to the `UmbracoApplicationBase.ApplicationInit` event. Here is an example:
+**BEWARE**! Handlers order is an important thing, and removing handlers or reordering handlers can have unexpected consequences.
 
-    using Umbraco.Core;
-    using Umbraco.Core.Events;
-    using Umbraco.Core.Logging;
-    using Umbraco.Core.Models;
-    using Umbraco.Core.Services;
-
-    namespace MyProject.EventHandlers
-    {
-        public class RegisterEvents : ApplicationEventHandler
-        {
-            protected override void ApplicationStarted(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
-            {
-                //Listen for the ApplicationInit event which then allows us to bind to the
-                //HttpApplication events.
-                UmbracoApplicationBase.ApplicationInit += UmbracoApplicationBase_ApplicationInit;     
-            }
-            
-            /// <summary>
-            /// Bind to the events of the HttpApplication
-            /// </summary>
-            void UmbracoApplicationBase_ApplicationInit(object sender, EventArgs e)
-            {
-                var app = (HttpApplication) sender;
-                app.PostRequestHandlerExecute += UmbracoApplication_PostRequestHandlerExecute;
-            }
-
-            /// <summary>
-            /// At the end of a handled request do something... 
-            /// </summary>            
-            void UmbracoApplication_PostRequestHandlerExecute(object sender, EventArgs e)
-            {
-                //Do something...
-            }
-        }
-    }
+## Related Links
+* [Troubleshooting Slow Startup](./Troubleshooting-Slow-Start)
```

Comment:
this needs to be .md I think?

Reply:
Uhm really?
Aren't the .md extentions messing up the links the deployed to OUR?  Or are they stripped automagically?